File: simplex.java/src/main/java/kr/co/tmon/simplex/actions/ParameterizedActionBinder.java
package kr.co.tmon.simplex.actions;

class ParameterizedActionBinder<TAction extends IParameterizedAction<TParam, TResult>, TParam, TResult> 
	extends ActionBinder<TAction, TResult>
	implements IParameterizedActionBinder<TAction, TParam, TResult> {

	public ParameterizedActionBinder(Class<TAction> clazz) throws InstantiationException, IllegalAccessException {
		super(clazz);
	}
}

File: Simplex.kotlin/src/main/kotlin/kr/co/tmon/simplex/actions/ParameterizedActionBinder.kt
package kr.co.tmon.simplex.actions

class ParameterizedActionBinder<TAction: IParameterizedAction<TParam, TResult>, TParam, TResult>(factory: () -> TAction)
    : IParameterizedActionBinder<TAction, TParam, TResult> {

    companion object {
        inline operator fun <reified TAction: IParameterizedAction<TParam, TResult>, TParam, TResult> invoke()
                = ParameterizedActionBinder<TAction, TParam, TResult> {
            TAction::class.java.newInstance()
        }
    }

    override val action: TAction = factory()
}